docs/development.txt

```diff
@@ -0,0 +1,84 @@
+process shall be followed. Significant work may include, but is not limited to:
+
+* A major new feature or subproject eg. a new storage adapter
+* Incrementing the version of a core dependency such as a Apache TinkerPop
+* Addition or deprecation of a public API
+* Internal shared data structure or API change
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the Github issue tracker
+* Start a DISCUSS thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called
+* Three +1 votes are required for the change to be accepted.
+
+[[branching]]
+Branching
+---------
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their
+own branches, and submitting pull requests when ready. If multiple developers wish to collaborate on a feature,
+they may request that a feature branch be created in JanusGraph repository.  If the developers are not committers, a
+JanusGraph committer will be assigned as the shepard for that branch.  The shepard will be responsible for merging
+pull requests to that branch.
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+Pull Requests
+-------------
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC - review-then-commit). This workflow is
+performed using Github's pull request process. In instances where the committer deems the full process
+unnecessary, commit-then-review (CTR) may be employed. This should be used infrequently and only for extremely trivial
+changes. Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.  The committer
+should also send a message to https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] to notify other
+project members of the CTR.  If another committer responds with a -1, the commit should be rolled back and the formal
+RTC process should be followed.
+
+Users wishing to contribute to JanusGraph should fork the https://github.com/janusgraph/janusgraph[JanusGraph]
+repository and then submit pull requests using the Github pull request process. Please include the relevant issue
+number in the title of your pull request. A vote (-1, 0, +1) is required before a pull request may be merged.
+Non-committers are free, and encouraged to vote. Their votes will be non-binding, but can be taken into consideration
+and are still very valuable community input. The following rules apply to the voting process.
+
+* Three +1 votes are required merge a pull request
```

Comment:
Is this standard practice for ASF projects to require three +1s for every commit? Is this not too many? I ask because a requirement of 3 reviewers with +1 may slow down the review / submit process for PRs. Thoughts?